vpr/src/device/rr_spatial_lookup.cpp

```diff
@@ -74,6 +74,88 @@ RRNodeId RRSpatialLookup::find_node(int x,
     return RRNodeId(rr_node_indices_[type][node_x][node_y][node_side][ptc]);
 }
 
+std::vector<RRNodeId> RRSpatialLookup::find_channel_nodes(int x,
+                                                          int y,
+                                                          t_rr_type type) const {
+    /* TODO: The implementation of this API should be worked 
+     * when rr_node_indices adapts RRNodeId natively!
+     */
+    std::vector<RRNodeId> channel_nodes;
+
+    /* Pre-check: the x, y, type are valid! Otherwise, return an empty vector */
+    if ((0 > x || 0 > y) && (CHANX == type || CHANY == type)) {
```

Comment:
@vaughnbetz Agree. I am impacted by the book [C traps and pitfalls](https://www.amazon.com/C-Traps-Pitfalls-Andrew-Koenig/dp/0201179288), which recommends to put constants on the left side. I guess they say so because at that time,  the compiler is not smart enough. I am cool now to follow the natural way in coding.